eno-core/src/main/java/fr/insee/eno/core/model/navigation/LinkedLoop.java

```diff
@@ -0,0 +1,89 @@
+    private String reference;
+
+    // TODO: method bindings -> no fully qualified classes in spel in annotations
+
+    /**
+     * In current DDI modeling, a loop object that is based on another loop does not directly have a reference towards
+     * the loop it is "based on".
+     * The link is done in variable group objects.
+     * This method contains the complex searching logic behind this.
+     * Note: a simple reference in the DDI loop object would avoid all this...
+     * TODO: As long as the DDI modeling is not simplified, dedicated business exception objects would be better. */
+    public static String findLoopReference(LoopType ddiLoop, DDIIndex index) {
+        String loopId = ddiLoop.getIDArray(0).getStringValue();
+        // Get the variable groups list
+        ControlConstructSchemeType controlConstructScheme = (ControlConstructSchemeType) index
+                .getParent(ddiLoop.getIDArray(0).getStringValue());
+        ResourcePackageType resourcePackage  = (ResourcePackageType) index
+                .getParent(controlConstructScheme.getIDArray(0).getStringValue());
+        List<VariableGroupType> variableGroupList = resourcePackage
+                .getVariableSchemeArray(0).getVariableGroupList();
+        // Find the variable group that contains a "BasedOnReference" object that matches the loop id
+        Optional<VariableGroupType> referenceGroup = variableGroupList.stream()
+                .filter(variableGroupType -> variableGroupType.getBasedOnObject().getBasedOnReferenceList()
+                        .stream()
+                        .map(referenceType -> referenceType.getIDArray(0).getStringValue())
+                        .anyMatch(loopId::equals))
+                .findAny();
+        if (referenceGroup.isEmpty()) {
+            throw new MappingException("Group associated with linked loop not found.");
+        }
+        // Using the variable group, we finally have the "based on" reference:
+        // it is the first "BasedOnReference" defined in the variable group
+        String referenceObjectId = referenceGroup.get().getBasedOnObject().getBasedOnReferenceArray(0)
+                .getIDArray(0).getStringValue();
+        if (loopId.equals(referenceObjectId)) {
+            throw new MappingException("Linked loop '"+loopId+"' referenced itself. " +
+                    "Note: Default implicit value is forbidden. " +
+                    "A loop must have either a minimum/maximum or a 'Based on' reference.");
+        }
+        // The linked loop is returned
+        AbstractIdentifiableType loopReference = index.get(referenceObjectId);
+        referenceTypeCheck(loopId, loopReference);
+        return referenceObjectId;
+    }
+
+    private static void referenceTypeCheck(String loopId, AbstractIdentifiableType loopReference) {
+        if (! (
+                loopReference instanceof LoopType ||
+                (loopReference instanceof QuestionGridType questionGridType
```

Comment:
I think it would be a good idea indeed. For now I insert this info myself in Lunatic objects, the way it is done is not well unified, so I prefer wait for the incoming changes in Lunatic-Model to see what will be made for this property.